horovod/ray/runner.py

```diff
@@ -305,35 +234,71 @@ def __init__(self,
         self.use_gpu = use_gpu
         self.gpus_per_slot = gpus_per_slot or 1
 
+        self.workers = []
+        self.placement_group = None
+
     @property
     def num_workers(self):
         return self.num_hosts * self.num_slots
 
     def _create_workers(self, host_resources):
-        colocator_cls = NodeColocator.options(**host_resources)
-        # Create a number of coordinators.
-        colocators = [
-            colocator_cls.remote(
-                node_rank=node_rank,
-                num_slots=self.num_slots,
-                world_size=self.num_workers,
-                use_gpu=host_resources["num_gpus"] > 0)
-            for node_rank in range(self.num_hosts)
-        ]
-        # We must save a pointer to each colocator to prevent their resource
-        # allocation from being released, along with their children from
-        # going out of scope.
-        self.colocators = colocators
-
-        node_ids = map_blocking(lambda a: a.create_workers.remote(),
-                                colocators)
-        if not len(set(node_ids)) == len(node_ids):
-            raise RuntimeError("Colocator actors must "
-                               f"be placed on unique nodes! Got: {node_ids}")
+        bundles = [host_resources.copy() for _ in range(self.num_hosts)]
+        pg = ray.util.placement_group(bundles, strategy="STRICT_SPREAD")
+        self.placement_group = pg
+        logger.debug("Waiting for placement group to start.")
+        ready, _ = ray.wait([pg.ready()],
+                            timeout=int(os.environ.get(
+                                "HOROVOD_PLACEMENT_GROUP_TIMEOUT_S", 100)))
```

Comment:
I think you can use the settings object for this instead of a separate env var

Reply:
Good point, I added it as a new field to MiniSettings